src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -0,0 +1,898 @@
+            set { this.maxReceivedMessageSize = value; }
+        }
+
+        internal bool DoNotEmitTrust
+        {
+            get { return this.doNotEmitTrust; }
+            set { this.doNotEmitTrust = value; }
+        }
+
+        internal XmlDictionaryReaderQuotas ReaderQuotas
+        {
+            get { return this.readerQuotas; }
+            set { this.readerQuotas = value; }
+        }
+
+        void GetSupportingTokensCapabilities(ICollection<SecurityTokenParameters> parameters, out bool supportsClientAuth, out bool supportsWindowsIdentity)
+        {
+            supportsClientAuth = false;
+            supportsWindowsIdentity = false;
+            foreach (SecurityTokenParameters p in parameters)
+            {
+                if (p.SupportsClientAuthentication)
+                    supportsClientAuth = true;
+                if (p.SupportsClientWindowsIdentity)
+                    supportsWindowsIdentity = true;
+            }
+        }
+
+        void GetSupportingTokensCapabilities(SupportingTokenParameters requirements, out bool supportsClientAuth, out bool supportsWindowsIdentity)
+        {
+            supportsClientAuth = false;
+            supportsWindowsIdentity = false;
+            bool tmpSupportsClientAuth;
+            bool tmpSupportsWindowsIdentity;
+            this.GetSupportingTokensCapabilities(requirements.Endorsing, out tmpSupportsClientAuth, out tmpSupportsWindowsIdentity);
+            supportsClientAuth = supportsClientAuth || tmpSupportsClientAuth;
+            supportsWindowsIdentity = supportsWindowsIdentity || tmpSupportsWindowsIdentity;
+
+            this.GetSupportingTokensCapabilities(requirements.SignedEndorsing, out tmpSupportsClientAuth, out tmpSupportsWindowsIdentity);
+            supportsClientAuth = supportsClientAuth || tmpSupportsClientAuth;
+            supportsWindowsIdentity = supportsWindowsIdentity || tmpSupportsWindowsIdentity;
+
+            this.GetSupportingTokensCapabilities(requirements.SignedEncrypted, out tmpSupportsClientAuth, out tmpSupportsWindowsIdentity);
+            supportsClientAuth = supportsClientAuth || tmpSupportsClientAuth;
+            supportsWindowsIdentity = supportsWindowsIdentity || tmpSupportsWindowsIdentity;
+        }
+
+        internal void GetSupportingTokensCapabilities(out bool supportsClientAuth, out bool supportsWindowsIdentity)
+        {
+            this.GetSupportingTokensCapabilities(this.EndpointSupportingTokenParameters, out supportsClientAuth, out supportsWindowsIdentity);
+        }
+
+        // SecureConversation needs a demuxer below security to 1) demux between the security sessions and 2) demux the SCT issue and renewal messages
+        // to the authenticator
+        internal void AddDemuxerForSecureConversation(ChannelBuilder builder, BindingContext secureConversationBindingContext)
+        {
+            // add a demuxer element  right below security unless there's a demuxer already present below and the only 
```

Comment:
What's the situation with this code? Is it no longer needed? Is this method still called?

Reply:
No we don't need this... we are no more relying BIndingElement to derive the demuxer logic... all logic is going through AddDemuxer (filter and Dispatcher), when the matching filter found, corresponding ChannelDispatcher and it's DispatchAsync called... removing the commented code... (not resolving, leaving for you to resolve)